### ISSUE TITLE
A problem occurred configuring project ':flutter_branch_sdk'.

### ISSUE THREAD

**Issue by heshesh2010** [RodrigoSMarques/flutter_branch_sdk#256]
run on new version 7.0.0
with android 

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':flutter_branch_sdk'.
> Could not create an instance of type com.android.build.api.variant.impl.LibraryVariantBuilderImpl.
   > Namespace not specified. Specify a namespace in the module's build file. See https://d.android.com/r/tools/upgrade-assistant/set-namespace for information about setting the namespace.

     If you've specified the package attribute in the source AndroidManifest.xml, you can use the AGP Upgrade Assistant to migrate to the namespace value in the build file. Refer to https://d.android.com/r/tools/upgrade-assistant/agp-upgrade-assistant for general information about using the AGP Upgrade Assistant.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

**Comment by RodrigoSMarques** [comment]
Does this occur on a new project or are you updating an existing project?

Did you read the instructions in the link provided? Does your project have those characteristics?

**Comment by crobinsonpicdmo** [comment]
I have the same problem. Support for the namespace in Gradle 8+ is needed
https://developer.android.com/build/configure-app-module#set-namespace

**Comment by chadpav** [comment]
After manually updating this package and several others I can confirm that supporting AGP 8+ is the correct solution. I guess all flutter packages are hitting this at the same time. In order to compile against SDK 34 I had to go to AGP 8.

I added this code to the `build.gradle` file:

```
    // Conditional for compatibility with AGP <4.2.
    if (project.android.hasProperty("namespace")) {
      namespace 'br.com.rsmarques.flutter_branch_sdk'
    }
```

**Comment by RodrigoSMarques** [comment]
@chadpav  and @crobinsonpicdmo.

Thank you for the informations.

In addition to including build.gradle, checking changes made to other plugins, there are other changes to be made.

I am mapping out what needs to be done and will make a new version available in the next few days.

**Comment by messinis** [comment]
Same issue here.

**Comment by RodrigoSMarques** [comment]
Resolved in version 7.0.1
